app/browser/tabs.js

```diff
@@ -8,14 +8,13 @@ const tabActions = require('../common/actions/tabActions')
 const config = require('../../js/constants/config')
 const Immutable = require('immutable')
 const tabState = require('../common/state/tabState')
-const windowState = require('../common/state/windowState')
 const {app, BrowserWindow, extensions, session, ipcMain} = require('electron')
 const {makeImmutable, makeJS} = require('../common/state/immutableUtil')
 const {getTargetAboutUrl, getSourceAboutUrl, isSourceAboutUrl, newFrameUrl, isTargetAboutUrl, isIntermediateAboutPage, isTargetMagnetUrl, getSourceMagnetUrl} = require('../../js/lib/appUrlUtil')
 const {isURL, getUrlFromInput, toPDFJSLocation, getDefaultFaviconUrl, isHttpOrHttps, getLocationIfPDF} = require('../../js/lib/urlutil')
 const {isSessionPartition} = require('../../js/state/frameStateUtil')
 const {getOrigin} = require('../../js/lib/urlutil')
-const {getSetting} = require('../../js/settings')
+const settingsStore = require('../../js/settings')
```

Comment:
Any specific reason for changing this? (just curious - I didn't see any other functions from settingsStore being called)

Reply:
@bsclifton I changed this so I could register a stub for `getSetting`. I couldn't see another way to do it unfortunately, but if there's a way to do it whilst preserving the ability to directly reference the child function of the `settingsStore` object in the module, I'd be keen to learn. The function is stubbed with the following:

```
+      this.stubGetSettings = sinon.stub(this.settings, 'getSetting',
+        (settingKey, settingsCollection, value) => {
+          if (settingKey === settings.TAB_CLOSE_ACTION) {
+            return tabCloseSettingValue
+          }
+          return false
+        }
+      )
+    })
```